utils/functionSignatures.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * This script prints the call signature of all the functions of an ABI
+ * It is a work in progress,to be used for creating a csv file of all call signatures that appear in this subgraph
+ */
+
+require('dotenv').config()
+const Protocol = require('@graphprotocol/graph-cli/src/protocols/')
+const toolbox = require('gluegun/toolbox');
+const { Command } = require('commander');
+const program = new Command();
+const fs = require('fs-extra')
+program.version('0.0.1');
+const keccak256 = require('keccak256')
+const protocolInstance = new Protocol('ethereum')
+const ABI = protocolInstance.getABI()
+
+async function main() {
+    const csvFile = './functionSignatures.csv'
+    fs.unlinkSync(csvFile)
+    fs.createFile(csvFile)
+    fs.readdir('./abisForSignatures', async (err, files) => {
+        files.forEach(async (file) => {
+            console.log(file)
+            const filePath = `abis/${file}`
+            const contractName = file.split('.json')[0]
+            await loadAbiFromFile(filePath).then(abi => {
+                if (abi.callFunctionSignatures()._tail !== undefined) {
+                    abi.callFunctionSignatures()._tail.array.forEach(item => {
+                        const functionName = item.slice(0, item.indexOf('('))
+                        const signature = `0x${keccak256(item).toString('hex').slice(0, 8)}`
+                        fs.appendFile(csvFile, `${contractName},${functionName},${signature}\n`)
+                    }
+                    )
+                }
+            })
+        });
+    });
+}
+
+
+const loadAbiFromFile = async filename => {
```

Comment:
we use this method on several different scripts
maybe we should extract it to external utils files to DRY the code a little bit